packages/webdriverio/src/middlewares.js

```diff
@@ -45,8 +45,33 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
             }).apply(this)
         }
 
-        return fn(commandName, commandFn).apply(this, args)
+        try {
+            return fn(commandName, commandFn).apply(this, args)
+        } catch (error ) {
+            if (error.message.includes(`stale element reference`)) {
+                let currentElement = this;
+
+                // Generate selector array
+                let selectors = [];
+                do {
+                    selectors.push(currentElement.selector);
+                    currentElement = currentElement.parent || {};
+                } while (currentElement.elementId);
+                // Iterator is currently set to Browser
+                selectors.reverse();
+
+                // Chain elements through the array
+                const newElement =  selectors.reduce((elementPromise, selector) =>
```

Comment:
elementPromise is of the same type as the selector here, so isn't this simply a string to locate the element?

Reply:
Updated to no longer use promises.